packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -51,25 +57,57 @@ export default function BlockToolbar( { hideDragHandle } ) {
 		};
 	}, [] );
 
-	if ( blockClientIds.length === 0 ) {
-		return null;
-	}
+	const containerNodeRef = useRef();
+	const nodeRef = useRef();
+
+	const {
+		showMovers,
+		gestures: showMoversGestures,
+	} = useShowMoversGestures( { ref: nodeRef } );
+
+	useExperimentalToolbarPositioning( { ref: containerNodeRef } );
 
 	const shouldShowVisualToolbar = isValid && mode === 'visual';
 	const isMultiToolbar = blockClientIds.length > 1;
 
+	const shouldShowMovers =
+		getIsSmallerViewport() || ( showMovers && hasMovers );
```

Comment:
We have a React hook for this `getIsSmallerViewport` it's called `useViewportMatch`

Reply:
Awesome! Thank you!! I refactored :D